### PR TITLE
For #5312 - Add Telemetry for Tracking Protection

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt [hunk 1]
```diff
@@ -113,6 +114,11 @@ sealed class Event {
     object PrivateBrowsingPinnedShortcutPrivateTab : Event()
     object PrivateBrowsingStaticShortcutTab : Event()
     object PrivateBrowsingStaticShortcutPrivateTab : Event()
+    object TrackingProtectionTrackerList : Event()
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: I don't see this event been use in another place, is it intentionally?

**Comment by ekager** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**
- How often users interact with tracking protection panel
- How often users have to add a site exception because of site problems
- How often users edit tracking protection settings

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

- Usefulness and discoverability of tracking protection panel
- How often users change the tracking protection settings to a lower level of protection

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

* We currently record if TP is on/off but we have more granular options now and more UI surrounding TP

4) **Can current instrumentation answer these questions?**
* Currently there are no events to track this.

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

* All data is Category 2.

6) **How long will this data be collected?**

* Until 03/01/2020

7) **What populations will you measure?**

* All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

* Glean / Amplitude

9) **Where do you intend to share the results of your analysis?**

* Only on Glean, Amplitude and with mobile teams.

File: app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt (hunk 1 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: Oops updating to use it now :)

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: :)

File: docs/metrics.md [hunk 2]
```diff
@@ -138,6 +138,12 @@ The following metrics are added to the ping:
 | sync_auth.sign_in |[event](https://mozilla.github.io/glean/book/user/metrics/event.html) |A user pressed the sign in button on the sync authentication page and was successfully signed in to FxA  |[1](https://github.com/mozilla-mobile/fenix/pull/2745#issuecomment-494918532)||2020-03-01 |
 | sync_auth.sign_out |[event](https://mozilla.github.io/glean/book/user/metrics/event.html) |A user pressed the sign out button on the sync account page and was successfully signed out of FxA  |[1](https://github.com/mozilla-mobile/fenix/pull/2745#issuecomment-494918532)||2020-03-01 |
 | sync_auth.sign_up |[event](https://mozilla.github.io/glean/book/user/metrics/event.html) |User registered a new Firefox Account, and was signed into it  |[1](https://github.com/mozilla-mobile/fenix/pull/4931#issuecomment-529740300)||2020-03-01 |
+| tracking_protection.etp_setting_changed |[event](https://mozilla.github.io/glean/book/user/metrics/event.html) |A user added a TP exception through the TP toggle in the panel.  |[1](https://github.com/mozilla-mobile/fenix/pull/TBD)|<ul><li>strict: The new setting for ETP: etp_strict, etp_standard</li></ul>|2020-03-01 |
```
Comment (sblatz) [inline docs/metrics.md]: Don't forget to update the "TBD" part of this with a data review 😄

Comment (ekager) [reply docs/metrics.md]: Oops just needed to rebuild and push :)

**Comment by sblatz** [comment]
This just has a `do not land` until @boek can do a data review :)

**Comment by sblatz** [comment]
@ekager did we ever hear back from @vesta0 about adding this to the "core ping"? I feel like it's actually more useful there.

**Comment by ekager** [comment]
I think we want to specifically know when users feel they need to change the default TP settings and how often they do that?

**Comment by liuche** [comment]
Make sure to regenerate md docs after changes too!

**Pull request merged**